keyboards/planck/keymaps/copface/keymap.c

```diff
@@ -0,0 +1,263 @@
+#include QMK_KEYBOARD_H
+#include "muse.h"
+
+// Custom Keycode Definitions
+#define LOWER MO(_LOWER)
+#define RAISE MO(_RAISE)
+#define META MO(_META)
+
+#define KC_PRS LGUI(LALT(LCTL(KC_4)))
+#define KC_EUR LALT(KC_4)
+#define KC_MGRV LALT(KC_GRV)
+
+#define KC_FORMAT LGUI(LALT(KC_L))
+#define KC_FBSP LALT(KC_BSPC)
+#define KC_FDEL LALT(KC_DEL)
+#define KC_PS3 LSFT(LCTL(LGUI(KC_3)))
+#define KC_PS4 LSFT(LCTL(LGUI(KC_4)))
+
+#define HK_TM LALT(KC_2)
+#define HK_AF9 LALT(KC_F9)
+
+#define KC_SRC LGUI(KC_F12)
+#define KC_RNME LSFT(KC_F6)
+
+#define M_LGSPC LGUI_T(KC_SPC)
+#define M_LCENT LCTL_T(KC_ENT)
+#define M_LGENT LGUI_T(KC_ENT)
+
+enum planck_layers {
+  _MAC = 0,
+  _WIN,
+  _LOWER,
+  _RAISE,
+  _META,
+  _ADJUST
+};
+
+enum planck_keycodes {
+  MAC = SAFE_RANGE,
+  WIN,
+  KC_ARR,  // ->
+  KC_FARR, // =>
+  HK_CLOS, // />
+  XD1,     // :D
+  XD3,     // :DDD
+  XD5,     // :DDDDD
+  XD15,    // :DDDDDDDDDDDDDDD
+};
+
+enum tapdancers {
+    T_BR = 0, // [, ]
+    T_PA, // (, )
+    T_CU, // {, }
+    T_DL, // $, €
+    T_SL, // slash, backslash
+};
+
+//Tap Dance Definitions
+qk_tap_dance_action_t tap_dance_actions[] = {
+  [T_BR] = ACTION_TAP_DANCE_DOUBLE(KC_LBRC, KC_RBRC),
+  [T_PA] = ACTION_TAP_DANCE_DOUBLE(KC_LPRN, KC_RPRN),
+  [T_CU] = ACTION_TAP_DANCE_DOUBLE(KC_LCBR, KC_RCBR),
+  [T_DL] = ACTION_TAP_DANCE_DOUBLE(KC_DLR, KC_EUR),
+  [T_SL] = ACTION_TAP_DANCE_DOUBLE(KC_SLSH, KC_BSLS),
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[_MAC] = LAYOUT_planck_grid(
+    KC_ESC,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,  KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+    KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,  KC_J,    KC_K,    KC_L,    XXXXXXX, KC_ENT,
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,  KC_M,    KC_COMM, KC_DOT,  XXXXXXX, KC_LEAD,
+    KC_LGUI, KC_LCTL, KC_LALT, KC_LALT, LOWER,   KC_SPC,  META,  RAISE,   KC_RALT, KC_MPRV, KC_MPLY, KC_MNXT
+),
+
+[_WIN] = LAYOUT_planck_grid(
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    KC_LCTL, KC_LGUI, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

Comment:
you don't need a whole different layer for this. 

You could use the `Control GUI Swap` bootmagic codes, instead.
https://docs.qmk.fm/#/feature_bootmagic?id=keycodes

These keycodes are still functional, even if bootmagic is disabled (but not if "magic" is disabled)

Also, setting this as the default layer .... will prevent you from using the _MAC layer.

Reply:
Thanks a lot for all of the input! The code looks way cleaner now.

One question about this particular thing: I made a single layer for the keyboard and added the `MAGIC_SWAP_LCTL_LGUI` (as `LCG_SWP`) to the `_ADJUST` layer, and I tried enabling the `COMMAND_ENABLE = yes`, then `BOOTMAGIC_ENABLE = full` and `BOOTMAGIC_ENABLE = lite` in `rules.mk`, but nothing worked.

I guess I'm using the thing wrong. Would that work only if I plugged in my keyboard while pressing the buttons I specify? (holding `_RAISE` + `_LOWER` + `P` in my case?)

Thanks for taking the time to look at this! Greatly appreciated.

Reply:
For the `LCG_SWP` keycode, you don't need bootmagic nor command enabled.  You just put it in the layer, and hit it.  Though, you may want to use `CG_TOGG` instead, as this toggles the behavior.   

There isn't really any feedback for this, unless you have audio enabled, and a song set for `CG_NORM_SONG` and `CG_SWAP_SONG`

Reply:
Amazing, the `CG_TOGG` worked like a charm!
Thanks so much. This is so much cleaner.

Then I guess this PR is done now, I did all the changes you requested.